### DISCUSSION TITLE
How to protect endpoints by URL prefix without using decorators?

### DISCUSSION THREAD

**Discussion by oddy** [pallets-eco/flask-security#988] [Q&A] [answered]
Hi, firstly great work you guys are doing on flask-sec, thank you so much!

Secondly, i have an app that is hosted in multiple different ways, only one of which we want to use with flask-sec. Currently its security policy uses URL prefixes to apply security to different parts of the application. 

How can i do this with modern flask-security? Do i have to replicate the functionality of e.g. decorators.auth_required?

**Comment by jwag956** [comment, marked as answer]
Not totally sure I understand what you are trying to do - however - 'decorators' are really just callable functions that return a callable.. so you can use them directly in code rather than as a 'decorator'.

```
@route("/prefix")
def myview():
    auth_required()(real_my_view_for_prefix)()
```
Or you could create your OWN decorator that performs different algorithms depending on what security framework you are using.
